### DISCUSSION TITLE
target 类似 package 中的 set_sourcedir 功能

### DISCUSSION THREAD

**Discussion by car520120** [xmake-io/xmake#3651] [Ideas]
### 你在什么场景下需要该功能？

workdir
     ----src
     ----3rd
     ----xmgr
     xmake.lua

3rd 中只放 github 拖下来的源码不写额外的工程文件

### 描述可能的解决方案

xmake.lua
```lua
add_subdirs("xmgr")
```

xmgr/xmake.lua
```lua
includes("zlib.lua","minizip.lua")
```

xmgr/zlib.lua
```lua
target("zlib")
   set_kind("static")
   set_sourcedir("../3rd/zlib/")
   add_files("*.c")
```

**Comment by waruqi** [comment]
1. 用 includes， 不要用 add_subdirs 
2. 如何连 cmake/autoconf/makefile 都没，自己走 xmake.lua port + includes

**Comment by car520120** [comment]
主要是想使用 target:set_sourcedir  的功能，减少路径拼接 和 includes   add_subdirs  都是没有关系的
如果有 set_sourcedir  就不用写成以下方式，有有一点麻烦 
```lua
local lib_home = xxx 
target("zlib")
   set_kind("static")
   add_files(lib_home   .. "/*.c")
   add_files(lib_home   .. xxx)
```
